integration/guessinggame/Guess.sol

```diff
@@ -0,0 +1,58 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+interface IERC20 {
+    function totalSupply() external view returns (uint256);
+    function balanceOf(address account) external view returns (uint256);
+    function allowance(address owner, address spender) external view returns (uint256);
+
+    function transfer(address recipient, uint256 amount) external returns (bool);
+    function approve(address spender, uint256 amount) external returns (bool);
+    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
+
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+}
+
+contract Guess {
+    address payable owner;
+    uint8 private target;
+    uint16 public guesses;
+    uint8 public size;
+    IERC20 public erc20;
+    address public tokenAddress;
+
+    modifier onlyOwner {
+        require(msg.sender == owner, "Only owner can call this function.");
+        _;
+    }
+
+    constructor(uint8 _size, address _tokenAddress) {
+        owner = payable(msg.sender);
+        size = _size;
+        erc20 = IERC20(_tokenAddress);
+    }
```

Comment:
Can we add in setNewTarget() here? Otherwise on instantiation there is no target and the first number to guess will be zero. Seems much better to set a random number as the first target.

Reply:
ok. will add.
```    constructor(uint8 _size, address _tokenAddress) {
        owner = payable(msg.sender);
        size = _size;
        erc20 = IERC20(_tokenAddress);
        setNewTarget();
    }
```